app/bootloader/main.cpp

```diff
@@ -112,12 +112,13 @@ namespace factory_reset {
 
 	auto getCounter() -> uint8_t
 	{
-		FileManagerKit::File file {internal::factory_reset_counter_path, "r"};
-
-		if (!file.is_open()) {
+		if (auto file_does_not_exist = !FileManagerKit::file_exists(internal::factory_reset_counter_path);
+			file_does_not_exist) {
```

Comment:
for this one and for all the others, I'm wondering if the `if-statement` is really easier to read with the `init-statement`

`init-statement` are useful when you need the variable inside the `if` clause without the need of an external variable:

```cpp
// without
auto condition = some_function();

if (condition > 10) {
	do_something_with(condition);
}

// with
if (auto condition = some_function(); condition > 10) {
	do_something_with(condition);
}
```

if our case we don't do anything with the variable.

furthermore, a lot of times, we don't want to know `if the file DOES exist` but `if the file DOES *NOT* exist`

In this case, it make sens to provide the user with a function to do just that:

```cpp
auto file_exists(path) -> bool {
	return ...
}

auto file_does_not_exist(path) -> bool {
	return !file_exists(path);
}
```
that can be used:

```cpp
if (file_does_not_exist(path)) {
	// do something
}
```